Sources/Noora/Utilities/TerminalText.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+
+/// Terminal text represents a piece of texts where some elements have semantics
+/// that are used to format the text when it's output to the user based on the terminal
+/// capabilities.
+public struct TerminalText: Equatable, Hashable {
+    public enum Component: Equatable, Hashable {
+        /// A string with no special semantics in the context of terminal text.
+        case raw(String)
+        /// A string that represents a system command (e.g. 'tuist generate')
+        case command(String)
+    }
+
+    /// Every component of the interpolated string.
+    var components: [Component]
```

Comment:
```suggestion
    let components: [Component]
```

I don't see this being mutated?